### PR TITLE
Shader stuff & related fixes

### PR DESCRIPTION
**Pull request by guusw**
- Fix :Feature parameter on Drawable/glTF 
- Add shader bitwise operators (L/RShift, And,Xor,Or)
- Fix validation for ShardsVar inside pure & add test cases

### CODE REVIEW

File: src/core/runtime.cpp [hunk 1]
```diff
@@ -978,12 +978,10 @@ void validateConnection(ValidationContext &ctx) {
         shards::arrayPush(data.shared, type);
       }
     }
-    if (!ctx.wire || !ctx.wire->pure) {
-      // and inherited
-      for (auto &info : ctx.inherited) {
-        for (auto &type : info.second) {
-          shards::arrayPush(data.shared, type);
-        }
+    // and inherited
```
Comment (sinkingsugar) [inline src/core/runtime.cpp:981]: this likely is no good? are we sure @guusw ?

File: src/core/shards/wires.cpp [hunk 2]
```diff
@@ -133,13 +133,15 @@ SHTypeInfo WireBase::compose(const SHInstanceData &data) {
...
   IterableExposedInfo shared(data.shared);
   IterableExposedInfo sharedCopy;
-  if (mode == RunWireMode::Detached && !capturing) {
-    // keep only globals
-    auto end = std::remove_if(shared.begin(), shared.end(), [](const SHExposedTypeInfo &x) { return !x.global; });
-    sharedCopy = IterableExposedInfo(shared.begin(), end);
-  } else {
-    // we allow Detached but they need to be referenced during warmup
-    sharedCopy = shared;
+  if (!wire->pure) {
```
Comment (sinkingsugar) [inline src/core/shards/wires.cpp:136]: rationale?

**Review by sinkingsugar — CHANGES_REQUESTED** [review]
Can you explain a bit more? I have a bad feeling about the changes but I saw you added more tests.... the thing is that even if pure works we risk to break all other things

**Review by sinkingsugar — APPROVED** [review]

File: src/core/runtime.cpp (hunk 1 above)
Comment (guusw) [reply src/core/runtime.cpp:981]: Will add comment to clarify

File: src/tests/pure.edn [hunk 3]
```diff
@@ -1,15 +1,45 @@
...
+    4 >= .value
+    (Do pure))
+  (prepare main))"
+  (EDN.Eval))
 
-(schedule root main)
-(if (run root 0.1) nil (throw "Root tick failed"))
\ No newline at end of file
+(schedule root failure-case)
+(if (run root 0.1) (throw "Expected failure") nil)
```
Comment (Kryptos-FR) [inline src/tests/pure.edn:45]: Smart

**Pull request merged**